scripts/docker-entrypoint.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash -x
+#
+# set up a user and switch to that user
+
+set -euo pipefail
+
+UID="${UID:-1000}"
+GID="${GID:-1000}"
+export USER=arm
+export HOME="/home/${USER}"
+
+echo "creating group [arm] with id ${GID}"
+groupadd -fo -g "${GID}" "${USER}"
```

Comment:
Is there some reason that these operations can't be done in the Dockerfile?  Based on my knowledge of Docker, it seems like it would be more efficient to set up users and groups, and import the config files during build time.

Reply:
it depends on how you're consuming this, as a Dockerfile you build into an image yourself or a Docker image you pull from a repo.
this method allows for the user/group to be specified at container start, which makes an image in a repo usable by anyone and for the user/group to be changed without requiring a new image.